datafusion/src/physical_plan/datetime_expressions.rs

```diff
@@ -268,6 +268,13 @@ pub fn to_timestamp(args: &[ColumnarValue]) -> Result<ColumnarValue> {
     )
 }
 
+/// now SQL function
+pub fn now(_: &[ColumnarValue]) -> Result<ColumnarValue> {
+    Ok(ColumnarValue::Scalar(ScalarValue::TimestampNanosecond(
+        Some(chrono::Utc::now().timestamp_nanos()),
```

Comment:
I think this needs access too to the `ExecutionProps` so it can use the query start timestamp here.

We can add a test that disables the optimizer rule(s) to show this still resolves to the same value.

Reply:
Thanks @Dandandan . I just started working on this. But finding it quite difficult to pass `ExecutionProps` to these functions. There are multiple types of `ExecutionPlan`. Do i need to add a reference of `ExecutionProps` to all of these?

Reply:
It seems to me that this function should never actually be called if the plan has been optimized (as this function call should have been replaced by a scalar function during constant folding).

Thus if this code is hit, wouldn't it be a bug? I can think of two possible behaviors:
1. The code we have now (generate a value, but not the intended one)
2. Return an error.

If we return an error it starts meaning we can't run plans without the optimizer which may not be ideal (I am not sure). 

Thus,  in terms of "being nice to users" I would suggest option 1 and add a `warn!` here saying that the plan hasn't been optimized. 

```suggestion
        // this function should have been replaced with a constant as part of optimization
        // it this code is being run it likely means 1) there is a bug in constant folding or
        // 2) the plan was not optimized prior to being run.
        warn!("now() function has not been optimized");
        Some(chrono::Utc::now().timestamp_nanos()),
```

Reply:
@alamb This function also will return the timestamp from `ExecutionProps` instead of dynamically generating it. I'm trying to make that change right now.